### DISCUSSION TITLE
Site locale syntax does not work with Jekyll 4

### DISCUSSION THREAD

**Discussion by quantrung9** [mmistakes/minimal-mistakes#2720] [Q&A] [answered]
## Environment
- Minimal Mistakes version: 4.20.2 (latest)
- Ruby gem version: ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c)
- Jekyll version: 4.1.1
- Git repository URL: https://github.com/toolownerforked/minimal-mistakes-on-jekyll-4
- Operating system: Ubuntu 20.04

## Expected behavior

When I set the locale to another language, such as France, it does not display France as expected, still in English, even I stopped and started the Jekyll server again.

I dug into the template a little bit and find out that this syntax:

```
{{ site.data.ui-text[site.locale].recent_posts | inspect }}
```

will return nill, instead of a locale string.

## Steps to reproduce the behavior

install minimal-mistakes on Jekyll 4 (starting from `jekyll new` command), then set locale to France like this: https://github.com/toolownerforked/minimal-mistakes-on-jekyll-4/commit/92596e79cbf7c46a3855b0d5de51417e20e3536f

**Comment by iBug** [comment, marked as answer]
Check [the docs](https://mmistakes.github.io/minimal-mistakes/docs/quick-start-guide/#migrating-to-gem-version):

> 
From `v4.5.0` onwards, the default language files are read-in automatically via the [`jekyll-data`](https://github.com/ashmaroli/jekyll-data) plugin **if it's installed**. For sites hosted with GitHub Pages, you still need to copy the [`_data/ui-text.yml`](https://github.com/mmistakes/minimal-mistakes/blob/master/_data/ui-text.yml) file because the `jekyll-data` plugin [is unsupported on GitHub Pages](https://docs.github.com/en/github/working-with-github-pages/about-github-pages-and-jekyll#plugins).

**Comment by quantrung9** [comment]
Thanks, @iBug, in my repo, I did not keep any copy of data files. When I mentioned the syntax and the template, I just want to show the issue behind that.

**Comment by mmistakes** [comment]
At some point I removed the jekyll-data plugin as a dependency. I had a reason but don't remember what it was.
But when it was removed that statement above was left in and probably should have been removed too as I don't think bundled `_data` files carry over into sites using the theme.

I could be wrong on that as I haven't tested. I seem to remember Jekyll core adding this as a feature and that could be why I removed the plugin, or not.

**Comment by iBug** [comment]
@quanengineering Sorry I didn't notice I copied the text from an older revision of the repository (possibly forgot to `git pull` or such). I've updated it to match the current version in the live documentation site.

To be clear: It's now recommended that you copy [`_data/ui-text.yml`](https://github.com/mmistakes/minimal-mistakes/blob/master/_data/ui-text.yml) into your repository for the localization feature to work.

**Comment by quantrung9** [comment]
Thanks @iBug, and @mmistakes.
